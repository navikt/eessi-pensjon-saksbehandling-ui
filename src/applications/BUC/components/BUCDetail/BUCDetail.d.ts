import { BucInfo, InstitutionNames } from "constants/types"

export interface BUCDetailProps {
  buc: Buc;
  bucInfo: BucInfo;
  className ?: string;
  institutionNames: InstitutionNames;
  locale: string;
  rinaUrl: string;
  t: (...args: any[]) => any
}
