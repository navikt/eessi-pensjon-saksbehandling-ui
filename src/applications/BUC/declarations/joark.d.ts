
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
