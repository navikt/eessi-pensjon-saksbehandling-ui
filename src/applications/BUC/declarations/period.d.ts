export interface PeriodDate {
  day: string;
  month: string;
  year: string;
}

export interface Country {
  label: string;
  value: string;
}

export interface Period {
  id?: number,
  type: string;
  startDate: PeriodDate;
  endDate?: PeriodDate;
  dateType: string;
  uncertainDate: boolean;
  country: Country;
  comment: string;
  otherType?: string;
  insuranceName?: string;
  insuranceType?: string;
  insuranceId ?: string;
  workActivity ?: string;
  workName ?: string;
  workType ?: string;
  workStreet ?: string;
  workCity ?: string;
  workZipCode ?: string;
  workRegion ?: string;
  childFirstName ?: string;
  childLastName ?: string;
  childBirthDate ?: PeriodDate;
  learnInstitution ?: string;
  payingInstitution ?: string;
  attachments ?: Array[any];
  [key: string]: any;
}

export type Periods = Array<Period>

export type PeriodErrors = {[k: string]: string | undefined}

export interface PayloadPeriod {
  land: string;
  periode: null | {
    lukketPeriode?: {
      fom: string | null;
      tom: string | null;
    }
    openPeriode?:  {
      fom: string | null;
      extra: string;
    }
  };
  vedlegg ?: Array[any];
  trygdeordningnavn ?: string;
  medlemskap ?: string;
  forsikkringEllerRegistreringNr ?: string;
  annenInformasjon ?: string;
  usikkerDatoIndikator ?: string;
  jobbUnderAnsattEllerSelvstendig ?: string;
  navnFirma ?: string;
  typePeriode ?: string;
  adresseFirma ?: {
    postnummer: string | undefined;
    by: string | undefined;
    land: string;
    gate: string | undefined;
    region: string | undefined;
  };
  navnPaaInstitusjon ?: string;
  informasjonBarn ?:  {
    etternavn: string | null | undefined;
    fornavn: string | null | undefined;
    foedseldato: string | null | undefined;
    land: string;
  }
}

export interface Payload {
  ansattSelvstendigPerioder ?: Array<PayloadPeriod>;
  boPerioder ?: Array<PayloadPeriod>;
  barnepassPerioder ?: Array<PayloadPeriod>;
  frivilligPerioder ?: Array<PayloadPeriod>;
  forsvartjenestePerioder ?: Array<PayloadPeriod>;
  foedselspermisjonPerioder ?: Array<PayloadPeriod>;
  opplaeringPerioder ?: Array<PayloadPeriod>;
  arbeidsledigPerioder ?: Array<PayloadPeriod>;
  sykePerioder ?: Array<PayloadPeriod>;
  andrePerioder ?: Array<PayloadPeriod>;
}

export interface P4000Info {
  person: any,
  bank: any,
  stayAbroad: Array<Period>
}

export interface P4000PayloadInfo {
  personInfo: any,
  bankInfo: any,
  periodeInfo: Array<PayloadPeriod>
}
