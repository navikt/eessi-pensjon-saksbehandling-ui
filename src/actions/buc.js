
const BUCLIST = {

  ONGOING: [
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ']
    },
    {
      type: 'P_BUC_02',
      name: 'UførePensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      comments: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz'],
      comments: ['foo', 'bar', 'baz']
    }
  ],
  OTHER: [
    {
      type: 'P_BUC_01',
      name: 'FIFOFA',
      dateCreated: 'dd.mm.åå',
      countries: ['RU', 'GB', 'CH']
    },
    {
      type: 'P_BUC_02',
      name: 'GRÅTASS',
      dateCreated: 'dd.mm.åå',
      countries: ['DE', 'TR', 'CA']
    }
  ]
}

const SEDS = [
  {
    name: 'P2000',
    status: 'sent',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P3000SE',
    status: 'draft',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P4000',
    status: 'received',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P5000',
    status: 'foo',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  }
]

export const fetchBucList = async () => {
  return BUCLIST
}

export const fetchSedListForBuc = async (buc) => {
  return SEDS
}
