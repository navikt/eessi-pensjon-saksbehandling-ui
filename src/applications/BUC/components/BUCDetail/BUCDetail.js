import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import moment from 'moment'
import { Nav } from 'eessi-pensjon-ui'
import InstitutionList from 'applications/BUC/components/InstitutionList/InstitutionList'
import './BUCDetail.css'

const BUCDetail = ({ buc, bucInfo, className, institutionNames, locale, rinaUrl, t }) => {
  return (
    <Nav.EkspanderbartpanelBase
      id='a-buc-c-bucdetail__panel-id'
      className={classNames('a-buc-c-bucdetail', 's-border', className)}
      apen
      heading={
        <Nav.Systemtittel
          id='a-buc-c-bucdetail__header-id'
          className='a-buc-c-bucdetail__header'
        >
          {buc.type + ' - ' + t('buc:buc-' + buc.type)}
        </Nav.Systemtittel>
      }
    >
      <div className='a-buc-c-bucdetail__body'>
        <dl className='a-buc-c-bucdetail__props'>
          <dt className='odd'>
            <Nav.Element>{t('ui:status')}:</Nav.Element>
          </dt>
          <dd className='odd' id='a-buc-c-bucdetail__props-status-id'>
            <Nav.Normaltekst>{t('ui:' + buc.status)}</Nav.Normaltekst>
          </dd>
          <dt>
            <Nav.Element>{t('buc:form-caseOwner')}:</Nav.Element>
          </dt>
          <dd id='a-buc-c-bucdetail__props-creator-id'>
            <InstitutionList
              t={t}
              institutions={[buc.creator]}
              institutionNames={institutionNames}
              locale={locale}
              type='joined'
            />
          </dd>
          <dt className='odd'>
            <Nav.Element>{t('ui:created')}:</Nav.Element>
          </dt>
          <dd className='odd' id='a-buc-c-bucdetail__props-startDate-id'>
            <Nav.Normaltekst>{moment(buc.startDate).format('DD.MM.YYYY')}</Nav.Normaltekst>
          </dd>
          <dt>
            <Nav.Element>{t('buc:form-rinaCaseNumber')}:</Nav.Element>
          </dt>
          <dd id='a-buc-c-bucdetail__props-caseId-id'>
            <Nav.Lenke
              id='a-buc-c-bucdetail__gotorina-link-id'
              className='a-buc-c-bucdetail__gotorina-link'
              href={rinaUrl + buc.caseId}
              target='rinaWindow'
            >
              {buc.caseId}
            </Nav.Lenke>
          </dd>
          <dt className='odd'>
            <Nav.Element>{t('ui:tags')}:</Nav.Element>
          </dt>
          <dd className='odd' id='a-buc-c-bucdetail__props-tags-id'>
            <Nav.EtikettLiten>{bucInfo && bucInfo.tags ? bucInfo.tags.map(tag => t('buc:' + tag)).join(', ') : ''}</Nav.EtikettLiten>
          </dd>
          <dt>
            <Nav.Element>{t('ui:comment')}:</Nav.Element>
          </dt>
          <dd id='a-buc-c-bucdetail__props-comment-id'>
            <Nav.Normaltekst>{bucInfo && bucInfo.comment ? bucInfo.comment : ''}</Nav.Normaltekst>
          </dd>
        </dl>
        <Nav.Undertittel
          id='a-buc-c-bucdetail__institutions-title-id'
          className='a-buc-c-bucdetail__institutions-title mb-2'
        >
          {t('buc:form-involvedInstitutions')}:
        </Nav.Undertittel>
        <div className='a-buc-c-bucdetail__institutions'>
          <InstitutionList
            t={t}
            institutions={buc.institusjon}
            institutionNames={institutionNames}
            locale={locale}
            type='joined'
          />
        </div>
      </div>
    </Nav.EkspanderbartpanelBase>
  )
}

BUCDetail.propTypes = {
  buc: PT.object.isRequired,
  bucInfo: PT.object.isRequired,
  className: PT.string,
  institutionNames: PT.object,
  locale: PT.string.isRequired,
  rinaUrl: PT.string.isRequired,
  t: PT.func.isRequired
}

export default BUCDetail
