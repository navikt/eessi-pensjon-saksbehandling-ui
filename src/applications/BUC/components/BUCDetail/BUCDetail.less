.a-buc-c-bucdetail {

  &__props {
    display: flex;
    flex-wrap: wrap;
    width: 100%;

    .odd {
      background-color: whitesmoke;
    }

    dt {
      width: 50%;
      padding-bottom: .25rem;
      padding-top: .25rem;

      .typo-element {
        margin-left: 0.5rem;
      }
    }

    dd {
      width: 50%;
      padding-bottom: .25rem;
      padding-top: .25rem;
      margin-bottom: 0;
    }
  }
}
