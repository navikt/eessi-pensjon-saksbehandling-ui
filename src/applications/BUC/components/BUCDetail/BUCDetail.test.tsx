import { Buc, BucInfo, BucsInfo } from 'declarations/buc'
import { mount, ReactWrapper } from 'enzyme'
import moment from 'moment'
import React from 'react'
import sampleBucs from 'resources/tests/sampleBucs'
import sampleBucsInfo from 'resources/tests/sampleBucsInfo'
import { stageSelector } from 'setupTests'
import BUCDetail, { BUCDetailProps, BUCDetailSelector } from './BUCDetail'

const defaultSelector: BUCDetailSelector = {
  locale: 'nb',
  rinaUrl: 'http://rinaurl.mock.com'
}

describe('applications/BUC/components/BUCDetail/BUCDetail', () => {
  let wrapper: ReactWrapper
  const buc: Buc = sampleBucs[0] as Buc
  const bucInfo: BucInfo = (sampleBucsInfo as BucsInfo).bucs['' + buc.caseId]
  const initialMockProps: BUCDetailProps = {
    buc: buc,
    bucInfo: bucInfo
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<BUCDetail {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.c-expandingpanel')).toBeTruthy()
    expect(wrapper.exists('.a-buc-c-bucdetail__body')).toBeTruthy()
    expect(wrapper.exists('.a-buc-c-bucdetail__props')).toBeTruthy()
    expect(wrapper.find('#a-buc-c-bucdetail__props-caseId-id').render().text()).toEqual(buc.caseId)
    expect(wrapper.find('#a-buc-c-bucdetail__props-creator-id').render().text()).toEqual('NAVAT07')
    expect(wrapper.find('#a-buc-c-bucdetail__props-startDate-id').render().text()).toEqual(moment(new Date(buc.startDate as number)).format('DD.MM.YYYY'))
    expect(wrapper.find('#a-buc-c-bucdetail__props-status-id').render().text()).toEqual('ui:' + buc.status)
    expect(wrapper.find('#a-buc-c-bucdetail__props-tags-id').render().text()).toEqual('buc:tag-vip')
    expect(wrapper.find('#a-buc-c-bucdetail__props-comment-id').render().text()).toEqual(bucInfo.comment)
    expect(wrapper.exists('.a-buc-c-bucdetail__institutions')).toBeTruthy()
    expect(wrapper.find('.a-buc-c-bucdetail__institutions').hostNodes().render().text()).toEqual(['NAVAT07', 'DEMO001', 'DEMO001', 'DEMO001'].join(''))
  })
})
