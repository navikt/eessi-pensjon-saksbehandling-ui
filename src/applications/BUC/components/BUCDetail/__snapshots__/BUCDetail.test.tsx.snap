// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/BUCDetail/BUCDetail Render: match snapshot 1`] = `
.c13 {
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  display: inline-block;
  line-height: 0px;
  margin: 0.25rem;
}

.c13.animate {
  opacity: 0.01;
  -webkit-animation: iPdXhn 0.5s forwards;
  animation: iPdXhn 0.5s forwards;
}

.c13.type-original.wave div {
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

.c13.type-circle.wave div img {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

ui-fss.g1[id="c12"] {
  content: "c13,";
}

.c15.type-circle {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid var(--navds-semantic-color-border);
  border-radius: 50%;
  overflow: hidden;
  position: relative;
}

.c15.type-circle img {
  position: absolute;
  margin: auto;
  top: -999px;
  left: -999px;
  bottom: -999px;
  right: -999px;
}

.c15.type-circle.size-XS {
  width: 16px;
  height: 16px;
  min-width: 16px;
  min-height: 16px;
  max-width: 16px;
  max-height: 16px;
}

.c15.type-circle.size-XS img {
  height: 18px;
}

.c15.type-circle.size-S {
  width: 24px;
  height: 24px;
  min-width: 24px;
  min-height: 24px;
  max-width: 24px;
  max-height: 24px;
}

.c15.type-circle.size-S img {
  height: 26px;
}

.c15.type-circle.size-M {
  width: 32px;
  height: 32px;
  min-width: 32px;
  min-height: 32px;
  max-width: 32px;
  max-height: 32px;
}

.c15.type-circle.size-M img {
  height: 35px;
}

.c15.type-circle.size-L {
  width: 40px;
  height: 40px;
  min-width: 40px;
  min-height: 40px;
  max-width: 40px;
  max-height: 40px;
}

.c15.type-circle.size-L img {
  height: 45px;
}

.c15.type-circle.size-XL {
  width: 48px;
  height: 48px;
  min-width: 48px;
  min-height: 48px;
  max-width: 48px;
  max-height: 48px;
}

.c15.type-circle.size-XL img {
  height: 54px;
}

.c15.type-circle.size-XXL {
  width: 72px;
  height: 72px;
  min-width: 72px;
  min-height: 72px;
  max-width: 72px;
  max-height: 72px;
}

.c15.type-circle.size-XXL img {
  height: 82px;
}

.c15.type-original.size-XS img {
  height: 16px;
  min-height: 16px;
  width: auto;
}

.c15.type-original.size-S img {
  height: 24px;
  min-height: 24px;
  width: auto;
}

.c15.type-original.size-M img {
  height: 32px;
  min-height: 32px;
  width: auto;
}

.c15.type-original.size-L img {
  height: 40px;
  min-height: 40px;
  width: auto;
}

.c15.type-original.size-XL img {
  height: 48px;
  min-height: 48px;
  width: auto;
}

.c15.type-original.size-XXL img {
  height: 72px;
  min-height: 72px;
  width: auto;
}

ui-fss.g2[id="c14"] {
  content: "c15,";
}

.c11 {
  padding: 0.25rem;
}

ui-fss.g5[id="c10"] {
  content: "c11,";
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9:not(.noMargin) {
  margin-bottom: 0.35rem;
}

ui-fss.g6[id="c8"] {
  content: "c9,";
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g7[id="c6"] {
  content: "c7,";
}

.c17 {
  margin-left: 0.5rem;
}

.c17.oneLine {
  white-space: nowrap;
}

ui-fss.g8[id="c16"] {
  content: "c17,";
}

.c5 {
  width: 50%;
  padding-bottom: 0.25rem;
  padding-top: 0.25rem;
  margin-bottom: 0;
}

ui-fss.g47[id="c4"] {
  content: "c5,";
}

.c3 {
  width: 50%;
  padding-bottom: 0.25rem;
  padding-top: 0.25rem;
}

.c3 .typo-element {
  margin-left: 0.5rem;
}

ui-fss.g48[id="c2"] {
  content: "c3,";
}

.c19 {
  padding: 0.5rem;
}

ui-fss.g49[id="c18"] {
  content: "c19,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c1 .odd {
  background-color: var(--navds-semantic-color-component-background-alternate);
}

ui-fss.g50[id="c0"] {
  content: "c1,";
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c15.type-original {
    display: inline-block;
  }

  .c15.type-original img {
    border: 1px solid var(--navds-semantic-color-border);
  }
}

@supports not (-ms-high-contrast:none) {
  .c15.type-original {
    border: 1px solid var(--navds-semantic-color-border);
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
  }
}

<div
  class="navds-panel navds-panel--border"
  data-testid="a_buc_c_BUCDetail"
  style="padding: 0px;"
>
  <div
    class="navds-accordion"
    data-testid="a_buc_c_BUCDetail--panel_id"
    style="border-radius: 4px;"
  >
    <div
      class="navds-accordion__item navds-accordion__item--open"
    >
      <button
        aria-expanded="true"
        class="navds-accordion__header navds-heading navds-heading--small"
        id="accordionContent-mock-uuid"
      >
        <h1
          class="navds-heading navds-heading--medium"
        >
          P_BUC_01 - buc:buc-P_BUC_01
        </h1>
        <svg
          aria-hidden="true"
          class="navds-accordion__expand-icon navds-accordion__expand-icon--flip"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 17L3 8.429 4.5 7l7.5 7.143L19.5 7 21 8.429 12 17z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="navds-accordion__expand-icon navds-accordion__expand-icon--filled navds-accordion__expand-icon--flip"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 17L3 8.429 5.5 6l6.5 6.5L18.5 6 21 8.429 12 17z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-labelledby="accordionContent-mock-uuid"
        role="region"
      >
        <div
          aria-hidden="false"
          class="ReactCollapse--collapse"
          style="height: auto; overflow: initial;"
        >
          <div
            class="ReactCollapse--content"
          >
            <div
              class="navds-accordion__content"
            >
              <dl
                class="c0 c1"
              >
                <dt
                  class="c2 c3 odd"
                >
                  <p
                    class="navds-label"
                  >
                    ui:status
                    :
                  </p>
                </dt>
                <dd
                  class="c4 c5 odd"
                  data-testid="a_buc_c_BUCDetail--status_id"
                >
                  <p
                    class="navds-body-long"
                  >
                    buc:status-open
                  </p>
                </dd>
                <dt
                  class="c2 c3"
                >
                  <p
                    class="navds-label"
                  >
                    buc:form-caseOwner
                    :
                  </p>
                </dt>
                <dd
                  class="c4 c5"
                  data-testid="a_buc_c_BUCDetail--creator_id"
                >
                  <div
                    class="c6 c7"
                  >
                    <div
                      class="c8 c9"
                      data-testid="a_buc_c_institutionlist--div-id"
                    >
                      <div
                        aria-hidden="true"
                        class="navds-popover navds-popover--hidden"
                        style="position: absolute; left: 0px; top: 0px;"
                        tabindex="-1"
                      >
                        <span
                          class="c10 c11"
                        >
                          Norge
                        </span>
                        <div
                          class="navds-popover__arrow"
                          data-popper-arrow="true"
                          style="position: absolute;"
                        />
                      </div>
                      <div
                        id="flag-f476487e-dead-4eb6-8e34-d0cd8854c5ba"
                        style="display: inline-block;"
                      >
                        <div
                          class="c12 c13 size-S type-circle animate"
                        >
                          <div
                            class="c14 c15 size-S type-circle"
                          >
                            <img
                              alt="Norge"
                              src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1100 800'%3e %3crect width='1100' height='800' fill='%23ef2b2d'/%3e %3crect width='200' height='800' x='300' fill='%23fff'/%3e %3crect width='1100' height='200' y='300' fill='%23fff'/%3e %3crect width='100' height='800' x='350' fill='%23002868'/%3e %3crect width='1100' height='100' y='350' fill='%23002868'/%3e %3c/svg%3e"
                            />
                          </div>
                        </div>
                      </div>
                      <p
                        class="c16 c17 navds-body-long"
                      >
                        NO:NAVAT07
                      </p>
                    </div>
                  </div>
                </dd>
                <dt
                  class="c2 c3 odd"
                >
                  <p
                    class="navds-label"
                  >
                    ui:created
                    :
                  </p>
                </dt>
                <dd
                  class="c4 c5 odd"
                  data-testid="a_buc_c_BUCDetail--startDate_id"
                >
                  <p
                    class="navds-body-long"
                  >
                    23.10.2019
                  </p>
                </dd>
                <dt
                  class="c2 c3"
                >
                  <p
                    class="navds-label"
                  >
                    buc:form-rinaCaseNumber
                    :
                  </p>
                </dt>
                <dd
                  class="c4 c5"
                  data-testid="a_buc_c_BUCDetail--caseId_id"
                >
                  <a
                    class="navds-link"
                    data-amplitude="buc.view.detail.rinaurl"
                    data-testid="a_buc_c_BUCDetail--gotorina_link_id"
                    href="http://rinaurl.mock.com600891"
                    target="rinaWindow"
                  >
                    600891
                  </a>
                </dd>
              </dl>
              <h1
                class="navds-heading navds-heading--small"
              >
                buc:form-involvedInstitutions
                :
              </h1>
              <div
                class="c18 c19"
                data-testid="a_buc_c_BUCDetail--institutions_id"
              >
                <div
                  class="c6 c7"
                  data-testid="a_buc_c_BUCDetail--institutionlist_id"
                >
                  <div
                    class="c8 c9"
                    data-testid="a_buc_c_institutionlist--div-id"
                  >
                    <div
                      aria-hidden="true"
                      class="navds-popover navds-popover--hidden"
                      style="position: absolute; left: 0px; top: 0px;"
                      tabindex="-1"
                    >
                      <span
                        class="c10 c11"
                      >
                        Norge
                      </span>
                      <div
                        class="navds-popover__arrow"
                        data-popper-arrow="true"
                        style="position: absolute;"
                      />
                    </div>
                    <div
                      id="flag-e19d102f-8df7-47d3-84d1-1f2b30c66dab"
                      style="display: inline-block;"
                    >
                      <div
                        class="c12 c13 size-S type-circle animate"
                      >
                        <div
                          class="c14 c15 size-S type-circle"
                        >
                          <img
                            alt="Norge"
                            src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1100 800'%3e %3crect width='1100' height='800' fill='%23ef2b2d'/%3e %3crect width='200' height='800' x='300' fill='%23fff'/%3e %3crect width='1100' height='200' y='300' fill='%23fff'/%3e %3crect width='100' height='800' x='350' fill='%23002868'/%3e %3crect width='1100' height='100' y='350' fill='%23002868'/%3e %3c/svg%3e"
                          />
                        </div>
                      </div>
                    </div>
                    <p
                      class="c16 c17 navds-body-long"
                    >
                      NO:NAVAT07
                    </p>
                  </div>
                </div>
                <div
                  class="c6 c7"
                  data-testid="a_buc_c_BUCDetail--institutionlist_id"
                >
                  <div
                    class="c8 c9"
                    data-testid="a_buc_c_institutionlist--div-id"
                  >
                    <div
                      aria-hidden="true"
                      class="navds-popover navds-popover--hidden"
                      style="position: absolute; left: 0px; top: 0px;"
                      tabindex="-1"
                    >
                      <span
                        class="c10 c11"
                      >
                        Frankrike
                      </span>
                      <div
                        class="navds-popover__arrow"
                        data-popper-arrow="true"
                        style="position: absolute;"
                      />
                    </div>
                    <div
                      id="flag-d8c2331a-201d-475f-8a6e-4560bbe4f012"
                      style="display: inline-block;"
                    >
                      <div
                        class="c12 c13 size-S type-circle animate"
                      >
                        <div
                          class="c14 c15 size-S type-circle"
                        >
                          <img
                            alt="Frankrike"
                            src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 3 2'%3e%3cpath fill='%23ED2939' d='M0 0h3v2H0z'/%3e%3cpath fill='%23fff' d='M0 0h2v2H0z'/%3e%3cpath fill='%23002395' d='M0 0h1v2H0z'/%3e%3c/svg%3e"
                          />
                        </div>
                      </div>
                    </div>
                    <p
                      class="c16 c17 navds-body-long"
                    >
                      FR:DEMOFR01
                    </p>
                  </div>
                </div>
                <div
                  class="c6 c7"
                  data-testid="a_buc_c_BUCDetail--institutionlist_id"
                >
                  <div
                    class="c8 c9"
                    data-testid="a_buc_c_institutionlist--div-id"
                  >
                    <div
                      aria-hidden="true"
                      class="navds-popover navds-popover--hidden"
                      style="position: absolute; left: 0px; top: 0px;"
                      tabindex="-1"
                    >
                      <span
                        class="c10 c11"
                      >
                        Storbritannia
                      </span>
                      <div
                        class="navds-popover__arrow"
                        data-popper-arrow="true"
                        style="position: absolute;"
                      />
                    </div>
                    <div
                      id="flag-24b740c3-577e-4aab-b821-9a4a5e75b38a"
                      style="display: inline-block;"
                    >
                      <div
                        class="c12 c13 size-S type-circle animate"
                      >
                        <div
                          class="c14 c15 size-S type-circle"
                        >
                          <img
                            alt="Storbritannia"
                            src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 60 30'%3e%3cclipPath id='a'%3e%3cpath d='M0 0v30h60V0z'/%3e%3c/clipPath%3e%3cclipPath id='b'%3e%3cpath d='M30 15h30v15zv15H0zH0V0zV0h30z'/%3e%3c/clipPath%3e%3cg clip-path='url(%23a)'%3e%3cpath d='M0 0v30h60V0z' fill='%23012169'/%3e%3cpath d='M0 0l60 30m0-30L0 30' stroke='%23fff' stroke-width='6'/%3e%3cpath d='M0 0l60 30m0-30L0 30' clip-path='url(%23b)' stroke='%23C8102E' stroke-width='4'/%3e%3cpath d='M30 0v30M0 15h60' stroke='%23fff' stroke-width='10'/%3e%3cpath d='M30 0v30M0 15h60' stroke='%23C8102E' stroke-width='6'/%3e%3c/g%3e%3c/svg%3e"
                          />
                        </div>
                      </div>
                    </div>
                    <p
                      class="c16 c17 navds-body-long"
                    >
                      GB:DEMOGB01
                    </p>
                  </div>
                </div>
                <div
                  class="c6 c7"
                  data-testid="a_buc_c_BUCDetail--institutionlist_id"
                >
                  <div
                    class="c8 c9"
                    data-testid="a_buc_c_institutionlist--div-id"
                  >
                    <div
                      aria-hidden="true"
                      class="navds-popover navds-popover--hidden"
                      style="position: absolute; left: 0px; top: 0px;"
                      tabindex="-1"
                    >
                      <span
                        class="c10 c11"
                      >
                        Storbritannia
                      </span>
                      <div
                        class="navds-popover__arrow"
                        data-popper-arrow="true"
                        style="position: absolute;"
                      />
                    </div>
                    <div
                      id="flag-b3adf8a7-6ec6-4b04-bebd-eb62e190b27b"
                      style="display: inline-block;"
                    >
                      <div
                        class="c12 c13 size-S type-circle animate"
                      >
                        <div
                          class="c14 c15 size-S type-circle"
                        >
                          <img
                            alt="Storbritannia"
                            src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 60 30'%3e%3cclipPath id='a'%3e%3cpath d='M0 0v30h60V0z'/%3e%3c/clipPath%3e%3cclipPath id='b'%3e%3cpath d='M30 15h30v15zv15H0zH0V0zV0h30z'/%3e%3c/clipPath%3e%3cg clip-path='url(%23a)'%3e%3cpath d='M0 0v30h60V0z' fill='%23012169'/%3e%3cpath d='M0 0l60 30m0-30L0 30' stroke='%23fff' stroke-width='6'/%3e%3cpath d='M0 0l60 30m0-30L0 30' clip-path='url(%23b)' stroke='%23C8102E' stroke-width='4'/%3e%3cpath d='M30 0v30M0 15h60' stroke='%23fff' stroke-width='10'/%3e%3cpath d='M30 0v30M0 15h60' stroke='%23C8102E' stroke-width='6'/%3e%3c/g%3e%3c/svg%3e"
                          />
                        </div>
                      </div>
                    </div>
                    <p
                      class="c16 c17 navds-body-long"
                    >
                      UK:DEMOUK01
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
